Client/Frontend/Browser/OpenSearch.swift

```diff
@@ -47,10 +61,11 @@ class OpenSearchEngine {
 
     private func getURLFromTemplate(searchTemplate: String, query: String) -> NSURL? {
         let allowedCharacters = NSCharacterSet(charactersInString: SearchTermsAllowedCharacters)
-
         if let escapedQuery = query.stringByAddingPercentEncodingWithAllowedCharacters(allowedCharacters) {
             let urlString = searchTemplate.stringByReplacingOccurrencesOfString("{searchTerms}", withString: escapedQuery, options: NSStringCompareOptions.LiteralSearch, range: nil)
-            return NSURL(string: urlString)
+            if let encodedUrlString = urlString.stringByAddingPercentEncodingWithAllowedCharacters(NSCharacterSet.symbollessURLAllowedSet()) {
```

Comment:
Actually, the symbolCharacterSet line doesn't matter because the union of the allowed URL character sets will automatically filter out any invalid UTF16/32 characters. I can safely remove that line. I'll rename the extension to be URLAllowedCharacterSet :+1: